File: election-ui-swift/Modules/ElectionResults/Sources/ElectionResults/Public/ResultsService.swift

import Foundation

/// Error descriptor used when fetching election results
public enum ResultsServiceError: Error {
    /// Represents an error decoding the JSON data retrieved by the service. The JSON data may be malformed, or doesn't match the expected structure.
    case invalidJSON
}

/// Provides up-to-date election results. For more information on creating a `ResultsService` instance, see `ResultsServiceFactory`
public protocol ResultsService {
    
    /// Fetch the latest results
    /// - Parameter completion: block called when a result has been retrieved containing the latest election results data.
    func latestResults(completion: @escaping (Result<ElectionResponse, ResultsServiceError>) -> Void)
}

File: election-ui-kotlin/api/src/main/java/uk/co/bbc/elections/api/ResultsService.kt
package uk.co.bbc.elections.api

/**
 * Provides up-to-date election results.
 * For more information on creating a `ResultsService` instance, see `ResultsServiceFactory`
 */
interface ResultsService {
    /**
     * Fetch the latest results.
     *
     *
     * @return A results object containing the latest election results data.
     *
     * @throws ResultsRepositoryException if there was an error retrieving or
     *                                    parsing the results data from the back end.
     */
    @Throws(ResultsRepositoryException::class)
    suspend fun latestResults(): Results

}

